src/main/tree_node_settings.cpp

```diff
@@ -0,0 +1,129 @@
+#include "duckdb/main/tree_node_settings.hpp"
+
+namespace duckdb {
+
+void TreeNodeSettings::SetMetrics(unordered_set<TreeNodeSettingsType> &n_metrics) {
+	this->metrics = n_metrics;
+}
+
+unordered_set<TreeNodeSettingsType> &TreeNodeSettings::GetMetrics() {
+	return metrics;
+}
+
+void TreeNodeSettings::SetCpuTime(double cpu_time) {
+	if (!SettingEnabled(TreeNodeSettingsType::CPU_TIME)) {
+		return;
+	}
```

Comment:
For these simple arithmetic operations, or like here, just assigning the value, I wonder if the `set` access + comparison + early-out is more expensive than just performing the assign operator and inlining the function call 🤔